src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/LogValueCollection.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// The <see cref="FormattedLogValues"/> object created by the framework <see cref="ILogger"/> extensions
+    /// (LogInformation(), for example) require that all key-value pairs be included in the string message
+    /// passed to the method. We'd like to have short strings with a subset of the data, so this class wraps
+    /// <see cref="FormattedLogValuesCollection"/> and allows us to use the same behavior, but with an additional
+    /// payload not included in the message.
+    /// </summary>
+    internal class FormattedLogValuesCollection : IReadOnlyList<KeyValuePair<string, object>>
```

Comment:
Note: I created this class to work around this limitation: https://github.com/aspnet/Logging/issues/533